midend/lib/Conversion/LowerLinalgToGemmini/LowerLinalgToGemmini.cpp

```diff
@@ -75,6 +75,167 @@ class MatmulLowering : public OpRewritePattern<linalg::MatmulOp> {
   std::string accType;
 };
 
+class Conv2DNhwcFhwcLowering
+    : public OpRewritePattern<linalg::Conv2DNhwcFhwcOp> {
+public:
+  explicit Conv2DNhwcFhwcLowering(MLIRContext *context, std::string accType)
+      : OpRewritePattern(context), accType(accType) {}
+  using OpRewritePattern<linalg::Conv2DNhwcFhwcOp>::OpRewritePattern;
+  LogicalResult matchAndRewrite(linalg::Conv2DNhwcFhwcOp convOp,
+                                PatternRewriter &rewriter) const override {
+    Value input = convOp.getInputs()[0];
+    Value kernel = convOp.getInputs()[1];
+    Value output = convOp.getOutputs()[0];
+    Location loc = convOp.getLoc();
+
+    MemRefType inputType = dyn_cast<MemRefType>(input.getType());
+    MemRefType kernelType = dyn_cast<MemRefType>(kernel.getType());
+    MemRefType outputType = dyn_cast<MemRefType>(output.getType());
+
+    Type kernelElemType = kernelType.getElementType();
+    Type outputElemType = outputType.getElementType();
+
+    ArrayRef<int64_t> inputShape = inputType.getShape();
+
+    DenseIntElementsAttr dilationsAttr = convOp.getDilationsAttr();
+    DenseIntElementsAttr stridesAttr = convOp.getStridesAttr();
+
+    size_t dilations = 1;
+    size_t strides = 1;
+    if (dilationsAttr)
+      dilations = (*dilationsAttr.begin()).getLimitedValue();
+    if (stridesAttr)
+      strides = (*stridesAttr.begin()).getLimitedValue();
+
+    if (inputShape[1] != inputShape[2]) // h, w
```

Comment:
The comment should be more explicit and should be placed above the if statement.